source/language/types.rst

```diff
@@ -583,6 +636,17 @@ Casting specifics
 The classical types are divided into the 'standard' classical types (bool, int, 
 uint, and float) that exist in languages like C, and the 'special' classical 
 types (bit, angle, duration, and stretch) that do not.
+The standard types follow rules that mimick those of C99 for `promotion and
+conversion <https://en.cppreference.com/w/c/language/conversion>`_ in mixed
+expressions and assignments. Like the C99 specification, all float values have
+higher rank than all integer types, and higher-precision types have higher rank
+than the same type with a lower precision. Standard and special classical types
+may only mix in expressions with operators defined for those mixed types,
```

Comment:
**suggestion**: Point folks to the portion of the spec where they can find these.

Reply:
Added in https://github.com/Qiskit/openqasm/pull/337/commits/8f455d32344aa1e3513944180224ab5e4bff1b91